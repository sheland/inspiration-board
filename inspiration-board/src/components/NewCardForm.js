import React, { Component } from 'react';
import PropTypes from 'prop-types';
import emoji from 'emoji-dictionary';
import './NewCardForm.css';

const EMOJI_LIST = ["", "heart_eyes", "beer", "clap", "sparkling_heart", "heart_eyes_cat", "dog"]

class NewCardForm extends Component {
  constructor(props) {
    super(props);

    this.state = {
      cardText: '',
      cardEmoji: '',
    };
  }

  resetState = () => {
    this.setState({

      cardText: '',
      cardEmoji: '',

    });
  }

  onFormChange = (event) => {
    const field = event.target.name;
    const value = event.target.value;


    const updatedState = {};
    updatedState[field] = value;
    this.setState(updatedState);
  }

  onSubmit = (event) => {
    console.log(event)
    event.preventDefault();
    const { cardText, cardEmoji } = this.state;

    if (cardText === '' && cardEmoji === '') return;

    console.log(event);
    this.props.addCardCallback(this.state);
    this.resetState();
  }

  render() {
    return (
      <form onSubmit={this.onSubmit} name="new-card-form" id="new-card-form" className="new-card-form">
        <h3 className="new-card-form__header">Add a New Card</h3>

        <div>
          <label className="new-card-form__form-label" htmlFor="cardText">Text</label>
          <input className="new-card-form__form-textarea" name="cardText" placeholder="Inspire!" onChange={this.onFormChange} value={this.state.cardText}/>
        </div>

        <div>
          <label className="new-card-form__form-label">Select Emoji:</label>
          <select
            className="new-card-form__form-select"
            name="cardEmoji"
            onChange={this.onFormChange}
            value={this.state.cardEmoji}
            >

            {EMOJI_LIST.map((emojiText, index) => {
              return <option
                key={index}
                value={`${emojiText}`}>{emoji.getUnicode(emojiText)}</option>
            })}
          </select>
        </div>

        <input className="btn btn-success new-card-form__form-button" type="submit" name="submit" value="Add Card" />
      </form>
    );
  }


}

NewCardForm.propTypes = {
  addCardCallback: PropTypes.func.isRequired,
};

export default NewCardForm;
