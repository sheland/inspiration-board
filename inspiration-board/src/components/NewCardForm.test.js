// src/components/test/NewPetForm.test.js
import React from 'react';
import NewCardForm from '../NewCardForm';
import { shallow } from 'enzyme';

describe('NewCardForm', () => {
  it('matches an existing snapshot', () => {
    const wrapper = shallow(
      <NewCardForm
        addCardCallback={() => {} }
        />);

    // Assert that it looks like the last snapshot
    expect(wrapper).toMatchSnapshot();
  });
});
