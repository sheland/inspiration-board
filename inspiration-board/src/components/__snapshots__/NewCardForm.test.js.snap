// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewCardForm matches an existing snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NewCardForm
    addCardCallback={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h3
          className="new-card-form__header"
        >
          Add a New Card
        </h3>,
        <div>
          <label
            className="new-card-form__form-label"
            htmlFor="cardText"
          >
            Text
          </label>
          <input
            className="new-card-form__form-textarea"
            name="cardText"
            onChange={[Function]}
            placeholder="Inspire!"
            value=""
          />
        </div>,
        <div>
          <label
            className="new-card-form__form-label"
          >
            Select Emoji:
          </label>
          <select
            className="new-card-form__form-select"
            name="cardEmoji"
            onChange={[Function]}
            value=""
          >
            <option
              value=""
            />
            <option
              value="heart_eyes"
            >
              😍
            </option>
            <option
              value="beer"
            >
              🍺
            </option>
            <option
              value="clap"
            >
              👏
            </option>
            <option
              value="sparkling_heart"
            >
              💖
            </option>
            <option
              value="heart_eyes_cat"
            >
              😻
            </option>
            <option
              value="dog"
            >
              🐶
            </option>
          </select>
        </div>,
        <input
          className="btn btn-success new-card-form__form-button"
          name="submit"
          type="submit"
          value="Add Card"
        />,
      ],
      "className": "new-card-form",
      "id": "new-card-form",
      "name": "new-card-form",
      "onSubmit": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Add a New Card",
          "className": "new-card-form__header",
        },
        "ref": null,
        "rendered": "Add a New Card",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <label
              className="new-card-form__form-label"
              htmlFor="cardText"
            >
              Text
            </label>,
            <input
              className="new-card-form__form-textarea"
              name="cardText"
              onChange={[Function]}
              placeholder="Inspire!"
              value=""
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Text",
              "className": "new-card-form__form-label",
              "htmlFor": "cardText",
            },
            "ref": null,
            "rendered": "Text",
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "new-card-form__form-textarea",
              "name": "cardText",
              "onChange": [Function],
              "placeholder": "Inspire!",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <label
              className="new-card-form__form-label"
            >
              Select Emoji:
            </label>,
            <select
              className="new-card-form__form-select"
              name="cardEmoji"
              onChange={[Function]}
              value=""
            >
              <option
                value=""
              />
              <option
                value="heart_eyes"
              >
                😍
              </option>
              <option
                value="beer"
              >
                🍺
              </option>
              <option
                value="clap"
              >
                👏
              </option>
              <option
                value="sparkling_heart"
              >
                💖
              </option>
              <option
                value="heart_eyes_cat"
              >
                😻
              </option>
              <option
                value="dog"
              >
                🐶
              </option>
            </select>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Select Emoji:",
              "className": "new-card-form__form-label",
            },
            "ref": null,
            "rendered": "Select Emoji:",
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <option
                  value=""
                />,
                <option
                  value="heart_eyes"
                >
                  😍
                </option>,
                <option
                  value="beer"
                >
                  🍺
                </option>,
                <option
                  value="clap"
                >
                  👏
                </option>,
                <option
                  value="sparkling_heart"
                >
                  💖
                </option>,
                <option
                  value="heart_eyes_cat"
                >
                  😻
                </option>,
                <option
                  value="dog"
                >
                  🐶
                </option>,
              ],
              "className": "new-card-form__form-select",
              "name": "cardEmoji",
              "onChange": [Function],
              "value": "",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": "0",
                "nodeType": "host",
                "props": Object {
                  "children": undefined,
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": "option",
              },
              Object {
                "instance": null,
                "key": "1",
                "nodeType": "host",
                "props": Object {
                  "children": "😍",
                  "value": "heart_eyes",
                },
                "ref": null,
                "rendered": "😍",
                "type": "option",
              },
              Object {
                "instance": null,
                "key": "2",
                "nodeType": "host",
                "props": Object {
                  "children": "🍺",
                  "value": "beer",
                },
                "ref": null,
                "rendered": "🍺",
                "type": "option",
              },
              Object {
                "instance": null,
                "key": "3",
                "nodeType": "host",
                "props": Object {
                  "children": "👏",
                  "value": "clap",
                },
                "ref": null,
                "rendered": "👏",
                "type": "option",
              },
              Object {
                "instance": null,
                "key": "4",
                "nodeType": "host",
                "props": Object {
                  "children": "💖",
                  "value": "sparkling_heart",
                },
                "ref": null,
                "rendered": "💖",
                "type": "option",
              },
              Object {
                "instance": null,
                "key": "5",
                "nodeType": "host",
                "props": Object {
                  "children": "😻",
                  "value": "heart_eyes_cat",
                },
                "ref": null,
                "rendered": "😻",
                "type": "option",
              },
              Object {
                "instance": null,
                "key": "6",
                "nodeType": "host",
                "props": Object {
                  "children": "🐶",
                  "value": "dog",
                },
                "ref": null,
                "rendered": "🐶",
                "type": "option",
              },
            ],
            "type": "select",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "btn btn-success new-card-form__form-button",
          "name": "submit",
          "type": "submit",
          "value": "Add Card",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
    ],
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3
            className="new-card-form__header"
          >
            Add a New Card
          </h3>,
          <div>
            <label
              className="new-card-form__form-label"
              htmlFor="cardText"
            >
              Text
            </label>
            <input
              className="new-card-form__form-textarea"
              name="cardText"
              onChange={[Function]}
              placeholder="Inspire!"
              value=""
            />
          </div>,
          <div>
            <label
              className="new-card-form__form-label"
            >
              Select Emoji:
            </label>
            <select
              className="new-card-form__form-select"
              name="cardEmoji"
              onChange={[Function]}
              value=""
            >
              <option
                value=""
              />
              <option
                value="heart_eyes"
              >
                😍
              </option>
              <option
                value="beer"
              >
                🍺
              </option>
              <option
                value="clap"
              >
                👏
              </option>
              <option
                value="sparkling_heart"
              >
                💖
              </option>
              <option
                value="heart_eyes_cat"
              >
                😻
              </option>
              <option
                value="dog"
              >
                🐶
              </option>
            </select>
          </div>,
          <input
            className="btn btn-success new-card-form__form-button"
            name="submit"
            type="submit"
            value="Add Card"
          />,
        ],
        "className": "new-card-form",
        "id": "new-card-form",
        "name": "new-card-form",
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Add a New Card",
            "className": "new-card-form__header",
          },
          "ref": null,
          "rendered": "Add a New Card",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <label
                className="new-card-form__form-label"
                htmlFor="cardText"
              >
                Text
              </label>,
              <input
                className="new-card-form__form-textarea"
                name="cardText"
                onChange={[Function]}
                placeholder="Inspire!"
                value=""
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Text",
                "className": "new-card-form__form-label",
                "htmlFor": "cardText",
              },
              "ref": null,
              "rendered": "Text",
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "new-card-form__form-textarea",
                "name": "cardText",
                "onChange": [Function],
                "placeholder": "Inspire!",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <label
                className="new-card-form__form-label"
              >
                Select Emoji:
              </label>,
              <select
                className="new-card-form__form-select"
                name="cardEmoji"
                onChange={[Function]}
                value=""
              >
                <option
                  value=""
                />
                <option
                  value="heart_eyes"
                >
                  😍
                </option>
                <option
                  value="beer"
                >
                  🍺
                </option>
                <option
                  value="clap"
                >
                  👏
                </option>
                <option
                  value="sparkling_heart"
                >
                  💖
                </option>
                <option
                  value="heart_eyes_cat"
                >
                  😻
                </option>
                <option
                  value="dog"
                >
                  🐶
                </option>
              </select>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Select Emoji:",
                "className": "new-card-form__form-label",
              },
              "ref": null,
              "rendered": "Select Emoji:",
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <option
                    value=""
                  />,
                  <option
                    value="heart_eyes"
                  >
                    😍
                  </option>,
                  <option
                    value="beer"
                  >
                    🍺
                  </option>,
                  <option
                    value="clap"
                  >
                    👏
                  </option>,
                  <option
                    value="sparkling_heart"
                  >
                    💖
                  </option>,
                  <option
                    value="heart_eyes_cat"
                  >
                    😻
                  </option>,
                  <option
                    value="dog"
                  >
                    🐶
                  </option>,
                ],
                "className": "new-card-form__form-select",
                "name": "cardEmoji",
                "onChange": [Function],
                "value": "",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": "0",
                  "nodeType": "host",
                  "props": Object {
                    "children": undefined,
                    "value": "",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "option",
                },
                Object {
                  "instance": null,
                  "key": "1",
                  "nodeType": "host",
                  "props": Object {
                    "children": "😍",
                    "value": "heart_eyes",
                  },
                  "ref": null,
                  "rendered": "😍",
                  "type": "option",
                },
                Object {
                  "instance": null,
                  "key": "2",
                  "nodeType": "host",
                  "props": Object {
                    "children": "🍺",
                    "value": "beer",
                  },
                  "ref": null,
                  "rendered": "🍺",
                  "type": "option",
                },
                Object {
                  "instance": null,
                  "key": "3",
                  "nodeType": "host",
                  "props": Object {
                    "children": "👏",
                    "value": "clap",
                  },
                  "ref": null,
                  "rendered": "👏",
                  "type": "option",
                },
                Object {
                  "instance": null,
                  "key": "4",
                  "nodeType": "host",
                  "props": Object {
                    "children": "💖",
                    "value": "sparkling_heart",
                  },
                  "ref": null,
                  "rendered": "💖",
                  "type": "option",
                },
                Object {
                  "instance": null,
                  "key": "5",
                  "nodeType": "host",
                  "props": Object {
                    "children": "😻",
                    "value": "heart_eyes_cat",
                  },
                  "ref": null,
                  "rendered": "😻",
                  "type": "option",
                },
                Object {
                  "instance": null,
                  "key": "6",
                  "nodeType": "host",
                  "props": Object {
                    "children": "🐶",
                    "value": "dog",
                  },
                  "ref": null,
                  "rendered": "🐶",
                  "type": "option",
                },
              ],
              "type": "select",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "btn btn-success new-card-form__form-button",
            "name": "submit",
            "type": "submit",
            "value": "Add Card",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
      ],
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
